### PR TITLE
SNC: 29 Apr

### PR DESCRIPTION
**Pull request by Northmoc**
Few more cards that don't need engine updates

Feel like there should be a cleaner way to do Rogues' Gallery (and All Suns' Dawn) but that may be for another day

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/corpse_explosion.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:Corpse Explosion
+ManaCost:1 B R
+Types:Sorcery
+A:SP$ DamageAll | Cost$ 1 B R ExileFromGrave<1/Creature> | NumDmg$ X | ValidCards$ Creature,Planeswalker | ValidDescription$ each creature and each planeswalker. | SpellDescription$ CARDNAME deals damage equal to the exiled card's power to each creature and each planeswalker.
+SVar:X:Exiled$CardPower
+DeckHas:Ability$Graveyard
+DeckHints:Ability$Discard|Mill|Sacrifice
+Oracle:As an additional cost to cast this spell, exile a creature card from your graveyard.\nCorpse Explosion deals damage equal to the exiled card's power to each creature and each planeswalker.
```
Comment (Agetian) [inline forge-gui/res/cardsfolder/upcoming/corpse_explosion.txt:8]: Probably for this to function for the AI, it needs:
SVar:AIPreference:ExileFromGraveCost$Creature
(or something along those lines)

**Comment by Agetian** [comment]
Looks good to me!
Think that AI Preference hint is a good idea?

**Comment by Northmoc** [comment]
> Looks good to me! Think that AI Preference hint is a good idea?

I was surprised it was needed since in this case it's just restating the cost, but if the AI wouldn't play it without the hint, then yes I think it's a good idea 😁

**Review by Agetian — APPROVED** [review]

**Comment by Agetian** [comment]
Yeah, the AI rarely (if at all) makes use of those costs unless you give it a priority list. You can tweak it, e.g., make it only choose low-CMC cards by using SVar:AIPreference:ExileFromGraveCost$Creature.cmcLE3 etc.

**Pull request merged**